### PR TITLE
More date/datetime operations on Series

### PR DESCRIPTION
**Pull request by guarilha**
- `expr_day_of_week` was not available, so I was getting a `:nif_not_loaded` error when using it with `mutate`. 
- added a `date and datetime ops` test to `data_frame_test.exs`
- added `S.month/1`
- added `S.year/1`
- added `S.hour/1`
- added `S.minute/1`

### CODE REVIEW

File: lib/explorer/series.ex [hunk 1]
```diff
@@ -4099,6 +4099,104 @@ defmodule Explorer.Series do
...
+      iex> s = Explorer.Series.from_list([~N[2023-01-15 00:00:00], ~N[2022-02-16 23:59:59.999999], ~N[2021-03-20 12:00:00], nil])
+      iex> Explorer.Series.hour(s)
+      #Explorer.Series<
+        Polars[4]
+        integer [0, 23, 12, nil]
+      >
+  """
+  @doc type: :datetime_wise
+  @spec hour(Series.t()) :: Series.t()
+  def hour(%Series{dtype: dtype} = series) when K.in(dtype, [:datetime]),
```
Comment (philss) [inline lib/explorer/series.ex:4174]:
```suggestion
  def hour(%Series{dtype: :datetime} = series),
```

File: lib/explorer/series.ex [hunk 2]
```diff
@@ -4099,6 +4099,104 @@ defmodule Explorer.Series do
...
+      iex> s = Explorer.Series.from_list([~N[2023-01-15 00:00:00], ~N[2022-02-16 23:59:59.999999], ~N[2021-03-20 12:00:00], nil])
+      iex> Explorer.Series.minute(s)
+      #Explorer.Series<
+        Polars[4]
+        integer [0, 59, 0, nil]
+      >
+  """
+  @doc type: :datetime_wise
+  @spec minute(Series.t()) :: Series.t()
+  def minute(%Series{dtype: dtype} = series) when K.in(dtype, [:datetime]),
```
Comment (philss) [inline lib/explorer/series.ex:4194]:
```suggestion
  def minute(%Series{dtype: :datetime} = series),
```

**Review by philss — APPROVED** [review]
LGTM! :+1:

**Comment by guarilha** [comment]
Forgot the `S.second/1`... think thats it for now.

**Review by philss — APPROVED** [review]
:ship:

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**